spring-pulsar/src/test/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainerTests.java

```diff
@@ -87,6 +94,76 @@ void basicDefaultConsumer() throws Exception {
 		pulsarClient.close();
 	}
 
+	@Test
+	void containerPauseAndResumeFeatureUsingWaitAndNotify() throws Exception {
+		HashSet<String> topics = new HashSet<>();
+		topics.add("containerPauseResumeWaitNotify-topic");
+		Map<String, Object> config = Map.of("topicNames", topics, "subscriptionName",
+				"containerPauseResumeWaitNotify-sub");
+		PulsarClient pulsarClient = PulsarClient.builder().serviceUrl(PulsarTestContainerSupport.getPulsarBrokerUrl())
+				.build();
+		DefaultPulsarConsumerFactory<String> pulsarConsumerFactory = new DefaultPulsarConsumerFactory<>(pulsarClient,
+				config);
+		PulsarContainerProperties pulsarContainerProperties = new PulsarContainerProperties();
+		pulsarContainerProperties.setMessageListener((PulsarRecordMessageListener<?>) (consumer, msg) -> {
+		});
+		pulsarContainerProperties.setSchema(Schema.STRING);
+		DefaultPulsarMessageListenerContainer<String> container = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, pulsarContainerProperties);
+
+		Lock reentrantLock = new ReentrantLock();
+		Condition lockCondition = reentrantLock.newCondition();
+
+		Lock spyLock = spy(reentrantLock);
```

Comment:
Let's keep an eye on any random failures from this test as I remember that we have had issues w/ the consumer tests that use the spy mocks.